app/gui_api.py

```diff
@@ -3,10 +3,20 @@
 
 class GuiApi:
 
-    def __init__(self, auth_svc, plugins):
-        self.auth_svc = auth_svc
+    def __init__(self, services, plugins):
+        self.services = services
+        self.caldera_config = services.get('utility_svc').get_config()
+        self.auth_svc = services.get('auth_svc')
         self.plugins = plugins
 
+    def get_attack_url(self, request=None):
+        if 'attack_svc' in self.services:
```

Comment:
i don't like this function inside the gui_api (or gui plugin for that matter). i think this change would be better suited to live in the att&ck hook.py file. in there, you can give the plugin an "address", say "/plugins/attack", and have the resolving result be the att&ck webpage(s). see me on this if that sounds confusing.